salt/client/__init__.py

```diff
@@ -1677,15 +1677,37 @@ def _prep_pub(self, tgt, fun, arg, tgt_type, ret, jid, timeout, **kwargs):
         """
         Set up the payload_kwargs to be sent down to the master
         """
-        if tgt_type == "nodegroup":
-            if tgt not in self.opts["nodegroups"]:
+
+        def _resolve_nodegroup(ng):
```

Comment:
I'd prefer to see this method put at the module level, or as a class instance method. When methods are put inline like this it makes unit testing them much harder.

Reply:
> I'd prefer to see this method put at the module level, or as a class instance method.

Done.

> When methods are put inline like this it makes unit testing them much harder.

Is there an implication here that you'd like to see a test that exercises this method directly?  Or is this only to make it more easily patchable if required by some other test?

Reply:
I don't think it's necessary to test that function, as long as it's being tested.

But in general, having functions at module level does allow for such things, and I feel certain I've seen cases in Salt where internal functions have grown beyond my screen size.

My personal approach is that unless there's a good reason to write an internal function, don't 😄